crates/yakui-vulkan/examples/demo.rs

```diff
@@ -3,15 +3,43 @@ use yakui::geometry::{UVec2, Vec2};
 use yakui_vulkan::*;
 
 use winit::{
-    event::{ElementState, Event, KeyEvent, WindowEvent},
-    event_loop::ControlFlow,
+    event::{ElementState, KeyEvent, WindowEvent},
     keyboard::{KeyCode, PhysicalKey},
 };
 use yakui::image;
 
 const MONKEY_PNG: &[u8] = include_bytes!("../../bootstrap/assets/monkey.png");
 const DOG_JPG: &[u8] = include_bytes!("../assets/dog.jpg");
 
+struct DemoApp;
+
+impl App for DemoApp {
+    fn render(&mut self, gui_state: &GuiState) {
+        use yakui::{column, label, row, text, widgets::Text, Color};
+        let (animal, texture): (&'static str, yakui::TextureId) = match gui_state.which_image {
+            WhichImage::Monkey => ("monkye", gui_state.monkey.into()),
+            WhichImage::Dog => ("dog haha good boy", gui_state.dog),
+        };
+        column(|| {
+            row(|| {
+                label("Hello, world!");
+
+                let mut text = Text::new(48.0, "colored text!");
+                text.style.color = Color::RED;
+                text.show();
+            });
+
+            text(96.0, format!("look it is a {animal}"));
+
+            image(texture, Vec2::new(400.0, 400.0));
+        });
+    }
+}
+
+trait App {
+    fn render(&mut self, gui_state: &GuiState);
+}
```

Comment:
This abstraction is unhelpful:
- In an example, there can only ever possibly be one implementation
- This library is intended for direct use in applications, which will also not need anything like this
- Hypothetical framework-style downstreams will need something much more complex than this

Please simplify.